src/main/sensors/gyro.c

```diff
@@ -982,6 +1019,11 @@ static FAST_CODE void gyroUpdateSensor(gyroSensor_t *gyroSensor, timeUs_t curren
     if (gyroConfig()->checkOverflow) {
         checkForOverflow(gyroSensor, currentTimeUs);
     }
+    
+    if (gyroConfig()->gyro_yaw_spin_threshold > 0) {
+        checkForYawSpin(gyroSensor, currentTimeUs);
+    }
```

Comment:
Should be wrapped in the `#ifdef USE_`.

Also need to define the interaction between overflow and yaw spin.

Reply:
Yes at the moment my intent is just that one or the other is chosen.  In my case for MPU I enable the spin code, but disable the overflow code.  For ICM I do the opposite.

You're right, if the user chooses both, I guess the spin code should stop running when the overflow code happens.  In practice at the PID level the overflow code currently takes precedence when it is active, effectively stopping the spin code from doing anything.  And when the overflow code is not active, it isn't active. 

I guess you're view is that there should be something that inhibits the yaw spin detection code in gyro.c whenever the overflow code activates.

Given the conditional defines associated with overflow code, I'd be grateful if you could update this pull with something that would achieve that outcome, or something equivalent.

I suppose it would be, within the yaw spin code, at the start, if gyroSensor->overflowDetected = true then do nothing.  But I don't know how best to code that given the conditionals.  Your help would be appreciated with this.

Reply:
Or in the first line that determines what the yaw spin detection state is:

```if (gyroSensor->yawSpinDetected && !(gyroSensor->overflowDetected))```

Maybe that would work?